clippy_lints/src/manual_float_methods.rs

```diff
@@ -141,18 +155,26 @@ impl<'tcx> LateLintPass<'tcx> for ManualFloatMethods {
                                 format!("!{local_snippet}.is_infinite()"),
                                 Applicability::MaybeIncorrect,
                             );
-                        }
+                        },
                     }
-                }
+                },
             );
         }
     }
 }
 
-fn is_infinity(bits: u128) -> bool {
-    bits == 0x7f80_0000 || bits == 0x7ff0_0000_0000_0000
+fn is_infinity(constant: &Constant<'_>) -> bool {
+    match constant {
+        Constant::F32(bits) => bits.to_bits() == 0x7f80_0000,
+        Constant::F64(bits) => bits.to_bits() == 0x7ff0_0000_0000_0000,
+        _ => false,
+    }
 }
 
-fn is_neg_infinity(bits: u128) -> bool {
-    bits == 0xff80_0000 || bits == 0xfff0_0000_0000_0000
+fn is_neg_infinity(constant: &Constant<'_>) -> bool {
+    match constant {
+        Constant::F32(bits) => bits.to_bits() == 0xff80_0000,
+        Constant::F64(bits) => bits.to_bits() == 0xfff0_0000_0000_0000,
```

Comment:
(Minor readability nit)

```suggestion
        Constant::F32(float) => float.to_bits() == 0xff80_0000,
        Constant::F64(float) => float.to_bits() == 0xfff0_0000_0000_0000,
```